data/console/test_java.sh

```diff
@@ -207,98 +237,79 @@ test_javaplugin_alternatives
+        else
+	    if echo $java_version | grep -q ibm; then
+	        if ! rpm -qa | grep 'java.*devel' | grep -q $dot_version; then
+		    echo "Warning: java compiler alternative not found for $java_version"
+		    echo "Reason : devel pkg is not installed, thus it is normal there is not javac"
+		    if zypper products -i | grep -q sle-sdk; then
+		        echo "Status : Error! The devel pkg should be installed. Check your repos!"
+		        exit 1
+		    else
+		        echo "Status : Accepted because there is no SDK installed in the SUT"
+		    fi
+		    echo "SKIP Testing $java_version"
+		    echo
+		    continue
+	        fi
+	    fi
+	    echo "Error: java compiler alternative not found for $java_version"
+	    exit 1
+        fi
+        # Test if there's an alternativ for javaplugin, and if yes, set it as the current used one
+        # So far, only java-ibm offers this
+        if [[ $(uname -m) -ne "s390x" ]]; then
+	    if echo $java_version | grep -q ibm; then
+	        if grep $dot_version $LIST_ALL_JAVAPLUGIN_ALTERNATIVES > /dev/null; then
+		    javaplugin=$(grep $dot_version $LIST_ALL_JAVAPLUGIN_ALTERNATIVES)
+		    update-alternatives --set javaplugin $javaplugin
+	        else
+		    echo "Error: java plugin alternative not found for $java_version"
+		    exit 1
+	        fi
+	    fi
         fi
-    fi
-
-    # Test version active (linked)
-    test_java_version_active
-    test_javac_version_active
-    check_version_active_vs_dot "$java_version_active" "java"
-    check_version_active_vs_dot "$javac_version_active" "javac"
-
-    # Compile Hello World
-    javac $HELLO_WORLD
-    rq=$?
-    if [ $rq -ne 0 ]; then
-        echo "Java Compiler failed"
-        exit 1
-    else
-        echo "Java Compiler: PASS"
-    fi
-
-    # JVM is unable to load the *.class file unless you are into
-    # the very same directory that the file exist
-    cd $DIR; java Hello > /dev/null
-    rq=$?
-    if [ $rq -ne 0 ]; then
-        echo "Java failed to run the program"
-        exit 1
-    else
-        echo "Java runtime: PASS"
-    fi
-
-    # Go back to previous directory
-    cd $PWD
 
-    # Delete the successfully compiled files: class and binary of Hello World
-    rm $HELLO.class
-    echo
-done
+        # Test version active (linked)
+        test_java_version_active
+        test_javac_version_active
+        check_version_active_vs_dot "$java_version_active" "java"
+        check_version_active_vs_dot "$javac_version_active" "javac"
+        #Compile Hello World
+        compile_hello_world
+    done
+else 
+    echo -e "\n-----------------------------------------------------------------------------------------------------------------"
+    echo "Skip java/javac/plugin update alternative and some tests because of read-only file system on transactional server"
+    echo "-----------------------------------------------------------------------------------------------------------------"
```

Comment:
Also add a reference to `jsc#SLE-9101` just in case somebody needs to check out later, the reason why update alternatives are skipped in transactional server.

Reply:
okay. added.